source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1415,6 +1416,36 @@ TEST(NiFakeServiceTests, NiFakeService_GetAnIviDanceWithATwistArray_CallsGetAnIv
     EXPECT_EQ(response.actual_size(), expected_size);
 }
 
+TEST(NiFakeServiceTests, NiFakeService_SetCustomTypeArray_CallsSetCustomTypeArray)
+{
+    nidevice_grpc::SessionRepository session_repository;
+    std::uint32_t session_id = create_session(session_repository, kTestViSession);
+    NiFakeMockLibrary library;
+    nifake_grpc::NiFakeService service(&library, &session_repository);
+    ViInt32 number_of_elements = 2;
+    CustomStruct cs_array[] = { { 5, 8.0 },{ 15 , 19.7 } };
+
+    EXPECT_CALL(library, SetCustomTypeArray(kTestViSession, number_of_elements, _))
+        .With(Args<2, 1>(ElementsAreArray({ FieldsAre(cs_array[0].structInt, cs_array[0].structDouble),
+            FieldsAre(cs_array[1].structInt, cs_array[1].structDouble) })))
```

Comment:
@SparkingSpork @jonlugoatx 
In this unit test I am trying to test the server handler implementation of an API which has an input parameter of custom type. I wanted a matcher for array of Structs here in the EXPECT_CALL, but couldnt find one. So I used a hackish way to do this by using `ElementsAreArray` and `FieldsAre`. This worked in Windows and Ubuntu, but it is giving build errors in Linux RT(as you can see in this PR build). The error says : ` no matching function for call to ‘UnpackStruct(const CustomStruct&)’`. I needed some help, do either of you have any suggestions for this?
Tag : @doshirohan

Reply:
My best guess is you're hitting something with the compiler differences.
- Our desktop Linux build uses gcc-9.3.0
- Our Linux RT build uses gcc-6.3.0
It looks like there are a lot of C++ standards changed in between those versions. This is a complete guess, but this [change for C++17](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0127r2.html) that was added to[ gcc-7 ](https://gcc.gnu.org/projects/cxx-status.html) seems related after reading through the full error:
```
/home/runner/work/grpc-device/grpc-device/third_party/gtest/googlemock/include/gmock/gmock-matchers.h:3075:35: error: no matching function for call to ‘UnpackStructImpl(const CustomStruct&, testing::internal::MakeIndexSequence<2ul>, int)’
     -> decltype((UnpackStructImpl)(t, MakeIndexSequence<I>{}, 0)) {
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/grpc-device/grpc-device/third_party/gtest/googlemock/include/gmock/gmock-matchers.h:2983:6: note: candidate: template<class T, long unsigned int ...I> decltype (std::tie((get<I>)(testing::internal::UnpackStructImpl::t)...)) testing::internal::UnpackStructImpl(const T&, testing::internal::IndexSequence<I ...>, int)
 auto UnpackStructImpl(const T& t, IndexSequence<I...>, int)
      ^~~~~~~~~~~~~~~~
/home/runner/work/grpc-device/grpc-device/third_party/gtest/googlemock/include/gmock/gmock-matchers.h:2983:6: note:   template argument deduction/substitution failed:
/home/runner/work/grpc-device/grpc-device/third_party/gtest/googlemock/include/gmock/gmock-matchers.h: In substitution of ‘template<class T, long unsigned int ...I> decltype (std::tie((get<I>)(testing::internal::UnpackStructImpl::t)...)) testing::internal::UnpackStructImpl(const T&, testing::internal::IndexSequence<I ...>, int) [with T = CustomStruct; long unsigned int ...I = {0ul, 1ul}]’:
/home/runner/work/grpc-device/grpc-device/third_party/gtest/googlemock/include/gmock/gmock-matchers.h:3075:35:   required by substitution of ‘template<long unsigned int I, class T> decltype (testing::internal::UnpackStructImpl(t, testing::internal::MakeIndexSequence<N>{}, 0)) testing::internal::UnpackStruct(const T&) [with long unsigned int I = 2ul; T = CustomStruct]’
```

I think your better approach would be to write a custom matcher instead of trying to force the custom matchers to work here. It took me a bit to understand what your test code was doing and just making a custom matcher for an array of the custom struct might be easier.

@reckenro, I think you looked at making a custom matcher previously. Any input or suggestions here?

Reply:
Yeah, I dabbled in custom matchers a little bit but ended up avoiding it. You can see their documentation for defining custom matchers [here](https://github.com/google/googletest/blob/master/docs/reference/matchers.md#defining-matchers). I think the idea there would be to have your own MATCHER_P2 to define an `ElementsAreCustomStructArray` or something.

Another way to do this may be to just add an operator in this test file to be
```
bool operator==(const CustomStruct& first, const CustomStruct& second) 
{
  return first.structInt == second.structInt && first.structDouble == second.structDouble;
}
```
And then you could just have the normal `ElementsAreArray(cs_array)` I believe. I would try this first before going the custom matcher route.

Reply:
Coming from C# background I thought the `==` operator needs to be added as part of `struct` definition itself. Yeah, if we could add them here and make `ElementsAreArray` work, that would be great.

We also tried to figure out how to use `MATCHER_P2`, but couldn't figure out a way to get hold of input args in the matcher. Since the call would be `.With(Args<2, 1>(ElementsAreStructArray(cs_array))`, I am assuming we would have two arguments (array and its length). Would they be just `arg[0]` and `arg[1]`. The page you linked has example where there is single arg being passed to matcher, and it is accessed via `arg`, but we didn't fine anything for multiple params.

Reply:
Yeah, hopefully the first one works.

I looked at the matchers again and I think I got it confused. I think it would be a `MATCHER_P` where you pass in the expected `cs_array`. Then there would be two args represented as a tuple so you'd use `std::get<0>(arg)` and `std::get<1>(arg)` for the array and size arguments. I think this is how that would work. I'm a bit rusty on my C++.

Reply:
Yes, the first approach worked